fhirclient/models/bundle.py

```diff
@@ -50,19 +50,35 @@ def __init__(self, jsondict=None, strict=True):
         Type `str`. """
         
         super(Bundle, self).__init__(jsondict=jsondict, strict=strict)
-    
+
+    def __iter__(self):
+        """ Makes the Bundle itself an iterator by returning an iterator over its entries. """
+        if self.entry is None:
+            self._entry_iter = iter([])
+        else:
+            self._entry_iter = iter(self.entry)
+        return self
+
+    def __next__(self):
+        """ Returns the next BundleEntry in the Bundle's entry list using the internal iterator. """
+        # return next(self._entry_iter)
+
+        if not hasattr(self, '_entry_iter'):
+            self.__iter__()
+        return next(self._entry_iter)
```

Comment:
Also... having read both PRs now - maybe adding direct iteration support to Bundle is not worth the effort? Like the code in `_util.py` in this PR could handle iterating on entries and we wouldn't need any fhir-parser changes after all?

That said! I think it's cool to have the direct iteration. But is a new direction for us - adding custom cool Python features to individual classes. Which is nice! But if you want to avoid that complexity here, I'd be happy with all this iteration logic just living as fhirsearch-specific logic.

Reply:
(I know I sent you down the path of fhir-parser - so this is probably annoying to hear - I am now just thinking that everyone is likely going to get their Bundles from fhirsearch anyway... I guess it shows up in other places...  So direct iteration is still useful... But I think in my head, the iteration would be the pagination itself - which is annoying and helpful for us to provide. But if the iteration is over the entries, and we do pagination all in fhirsearch, maybe this bit of synctatic sugar can be left off for now, since it reduces the PR complexity a fair bit.)

Reply:
And to be clear, iterating over the entries makes sense too - my initial thought was over the pages, but entries feels more natural `for entry in bundle:`. So I'm not advocating for changing the iteration to pagination. I'm just realizing that iterating over entries is a small bit of syntactic sugar instead of a big one.